pkg/storage/bucket/object_client_adapter.go

```diff
@@ -103,26 +102,27 @@ func (o *ObjectClientAdapter) List(ctx context.Context, prefix, delimiter string
 
 	// If delimiter is empty we want to list all files
 	if delimiter == "" {
-		iterParams = append(iterParams, objstore.WithRecursiveIter)
+		iterParams = append(iterParams, objstore.WithRecursiveIter())
 	}
+	iterParams = append(iterParams, objstore.WithUpdatedAt())
 
-	err := o.bucket.Iter(ctx, prefix, func(objectKey string) error {
+	err := o.bucket.IterWithAttributes(ctx, prefix, func(attrs objstore.IterObjectAttributes) error {
```

Comment:
let's also check if the provider supports `UpdatedAt` option, if not we should make a follow-up Attributes call to fetch it given the callers of `List` might use the `ModifiedAt` field